plugins/statuslines/lualine.nix

```diff
@@ -0,0 +1,129 @@
+{ pkgs, config, lib, ... }:
+with lib;
+let
+  cfg = config.programs.nixvim.plugins.lualine;
+  helpers = import ../helpers.nix { lib = lib; };
+  separators = mkOption {
+    type = types.submodule {
+      options = {
+        left = mkOption {
+          default = " ";
+          type = types.str;
+          description = "left separator";
+        };
+        right = mkOption {
+          default = " ";
+          type = types.str;
+          description = "right separator";
+        };
+      };
+    };
+    default = { };
+  };
+  component_options = mode:
+    mkOption {
+      type = types.submodule {
+        options = {
+          mode = mkOption {
+            type = types.str;
+            default = "${mode}";
+          };
+          icons_enabled = mkOption {
+            type = types.enum [ "True" "False" ];
+            default = "True";
+            description = "displays icons in alongside component";
+          };
+          icon = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "displays icon in front of the component";
+          };
+          separator = separators;
+        };
+      };
+      default = { };
+    };
+in {
+  options = {
+    programs.nixvim.plugins.lualine = {
+      enable = mkEnableOption "Enable airline";
+
+      options = mkOption {
+        type = types.submodule {
+          options = {
+            theme = mkOption {
+              default = "auto";
+              type = types.str;
+              description = "The theme to use for lualine-nvim.";
+            };
+            section_separators = separators;
```

Comment:
I'll also rename these so that they're camelCase, more in line with the rest of the project